cli/lib/search.go

```diff
@@ -47,30 +52,58 @@ var searchFlags struct {
 }
 
 func runSearchCommand(cmd *cobra.Command, args []string) {
+	instance := cli.CreateInstance()
 	logrus.Info("Executing `arduino lib search`")
-	query := strings.ToLower(strings.Join(args, " "))
-
-	lm := cli.InitLibraryManager(cli.Config)
-
-	res := output.LibSearchResults{
-		Libraries: []*librariesindex.Library{},
+	arguments := strings.ToLower(strings.Join(args, " "))
+	outputsrc, err := lib.LibrarySearch(context.Background(), &rpc.LibrarySearchReq{
+		Instance: instance,
+		Names:    searchFlags.names,
+		Query:    arguments,
+	})
+	if err != nil {
+		formatter.PrintError(err, "Error saerching for Library")
+		os.Exit(cli.ErrGeneric)
 	}
-	for _, lib := range lm.Index.Libraries {
-		if strings.Contains(strings.ToLower(lib.Name), query) {
-			res.Libraries = append(res.Libraries, lib)
+	if cli.OutputJSONOrElse(outputsrc.GetSearchOutput()) {
```

Comment:
You're using `outputsrc.GetSearchOutput()` plenty of times, just assign it to a variable like:

```go
results := outputsrc.GetSearchOutput()
```

and use `results` instead

Reply:
done